app/controllers/katello/concerns/smart_proxies_controller_extensions.rb

```diff
@@ -0,0 +1,45 @@
+module Katello
+  module Concerns
+    module SmartProxiesControllerExtensions
+      extend ActiveSupport::Concern
+
+      included do
+        append_view_path('app/views/foreman')
+        before_filter :find_resource_and_status, :only => [:pulp_storage, :pulp_status]
+        alias_method_chain :action_permission, :katello
+
+        def pulp_storage
```

Comment:
@ehels API ATM is secondary, (phase two) one of the main reason its split
into its own service class is to reuse via the api controllers later on.